library/src/main/java/com/pokegoapi/util/BosslandHash.java

```diff
@@ -0,0 +1,125 @@
+            requests = new ArrayList<>();
+        }
+
+        public String json() {
+            Base64.Encoder encoder = Base64.getEncoder();
+            String auth = encoder.encodeToString(authTicket);
+            String session = encoder.encodeToString(sessionData);
+            String request = "";
+            for (byte[] array : requests)
+                request += "\"" + encoder.encodeToString(array) + "\",";
+            request = request.substring(0, request.length() - 1);
+
+            return "{"
+                    + "\"Timestamp\":\"" + timestamp + "\","
+                    + "\"Latitude\":" + latitude + ","
+                    + "\"Longitude\":" + longitude + ","
+                    + "\"Altitude\":" + altitude + ","
+                    + "\"AuthTicket\":\"" + auth + "\","
+                    + "\"SessionData\":\"" + session + "\","
+                    + "\"Requests\":[" + request +  "]"
+                    + "}";
+        }
+    }
+
+    private static String version(OkHttpClient client) throws IOException {
+        Response response = client.newCall(new Request.Builder()
+                .url(BASE + VERSION_END)
+                .build())
+                .execute();
+
+        if (response.code() == 200) { // OK
+            String version = response.body().string();
+            response.close();
+            return version.substring(version.lastIndexOf(":") + 2, version.length() - 2);
+        } else return null;
+    }
+
+    public static HashResponseContent request(PokemonGo api, OkHttpClient client, HashRequestContent content) throws IOException {
+        if (api.getHashKey() == null)
+            throw new NullPointerException("PokemonGo#getHashKey() returned null");
+
+        if (version == null) {
+            version = version(client);
+
+            if (version == null)
+                throw new IllegalStateException("Failed to load latest hashing version");
+        }
+
+        Call call = client.newCall(new Request.Builder()
+                .url(BASE + version)
+                .addHeader("Accept", "application/json")
+                .addHeader("X-AuthToken", api.getHashKey())
+                .post(RequestBody.create(MediaType.parse("application/json"), content.json()))
+                .build());
+        Response response = call.execute();
+
+        int code = response.code();
+        switch (code) {
+            case 200: // OK
+                String result = response.body().string();
+                response.close();
+                return new HashResponseContent(result);
+            case 400: // Bad request
+                throw new IllegalStateException("Bad request to hashing server, request=" + content.json());
+            case 401: // Unauthorized
+                throw new IllegalStateException("Unauthorized to request hash values");
+            case 429: // Limit exceeded
+                throw new IllegalStateException("Exceeded hashing limit");
```

Comment:
It probably should, this was just a raw convert to get it working. If noone else does it, I'll gladly improve on it when I get back home